### DISCUSSION TITLE
Exclude HTML on exportable column

### DISCUSSION THREAD

**Discussion by LaMance22** [Power-Components/livewire-powergrid#556] [Q&A]
Hi, thanks for this project ;). Please give me a suggestion. Can I exclude HTML when exporting columns?

I mark expiring date in red color. Its my code:

addColumn

`            ->addColumn('date_expire_formatted', function (Training $model) {
                $date_expire= $model->date_expire? Carbon::parse($model->date_expire)->format('d/m/Y') : NULL;
                if ( $model->date_expire< date('Y-m-d') ) {
                    return '<span class="text-gray-300">'.$date_expire.'</span>';
                } elseif ( $model->date_expire<= date('Y-m-d', strtotime('+ 10 days')) ) {
                    return '<span class="text-red-500 font-bold">'.$date_expire.'</span>';
                } else {
                    return $date_expire;
                }
            })
`

Column::make (date_expire is column name in database)

`
        Column::make(__('Date expire'), 'date_expire_formatted', 'date_expire')
        ->searchable()
        ->sortable(),
`

In excel:

![excel](https://user-images.githubusercontent.com/49619616/184045043-734c9657-68fe-4a39-b7bd-d304ae0d6b52.jpg)

**Comment by dansysanalyst** [comment]
Hi @LaMance22,

I am using [PowerGrid Demo](https://github.com/Power-Components/powergrid-demo) repository for the following example:

I have added two columns: First one with the formatted date containing HTML and one "clean", just the formatted date.

```php 
    public function addColumns(): PowerGridEloquent
    {
        return PowerGrid::eloquent()
             ->addColumn('id')    
            //...
            ->addColumn('produced_at_formatted', function (Dish $dish) {
                return '<b>' . Carbon::parse($dish->produced_at)->format('d/m/Y') . '</b>';
            })
            ->addColumn('produced_at_clean', function (Dish $dish) {
                return Carbon::parse($dish->produced_at)->format('d/m/Y');
            });
```

Next, I set `produced_at_clean` to be hidden on the screen but visible in export.

Then, `produced_at_formatted` is set to be hidden (not visible) in export.
 
```php 
    /**
    * PowerGrid Columns.
    *
    * @return array<int, Column>
    */
    public function columns(): array
    {
        return [
           //..
            Column::add()
                ->title(__('Production without HTML'))
                ->field('produced_at_clean')
                ->hidden()
                ->visibleInExport(true),

            Column::add()
                ->title(__('Production date'))
                ->field('produced_at_formatted')
                ->makeInputDatePicker('produced_at')
                ->visibleInExport(false)
        ];
```

This results in the column with HTML being displayed on the screen and the clean column to be hidden on the screen...


<img width="1384" alt="CleanShot 2022-08-11 at 22 36 53@2x" src="https://user-images.githubusercontent.com/79267265/184237701-bcb8cfe7-1c4b-4639-899a-b5d5084fd968.png">


And the opposite in the Excel file...

<img width="894" alt="CleanShot 2022-08-11 at 22 39 26@2x" src="https://user-images.githubusercontent.com/79267265/184238136-69d86076-1981-4f00-bbd8-f4beca8388c1.png">
